### PR TITLE
Remove references to older versions of the ResStock workflow

### LINKED ISSUE #1250: Remove references to old versions in README?
[The README](https://github.com/NREL/resstock/blob/develop/README.md) still provides some links/instructions for [ResStock 2.2.5](https://github.com/NREL/resstock/releases/tag/v2.2.5) (almost 4 years old, requires PAT to run) and [ResStock 2.5.0](https://github.com/NREL/resstock/releases/tag/v2.5.0) (over 2 years old, prior to OS-HPXML). Do we really want users to potentially use these old versions?

(A user that _really_ needs one of these older versions can still get it, it just wouldn't be presented as an option in the README.)

### PR DESCRIPTION
**Pull request by afontani**
## Pull Request Description

In the README.md, remove older versions of the ResStock workflow.

Resolves #1250. 

## Checklist

Not all may apply:

- [ ] Tests (and test files) have been updated
- [x] Documentation has been updated
  - [ ] If related to resstock-estimation, checklist includes [data dictionary](https://github.com/NREL/resstock/tree/develop/resources/data/dictionary), [source report](https://github.com/NREL/resstock/tree/develop/project_national/resources/source_report.csv), [options saturation](https://github.com/NREL/resstock/tree/develop/project_national/resources/options_saturations.csv), [options_lookup](https://github.com/NREL/resstock/blob/develop/resources/options_lookup.tsv).
- [ ] Changelog has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,15 +12,10 @@ This repository contains:
...
-- Project definition files:
-  - v2.3.0 and later: [buildstockbatch YML files openable in any text editor](https://github.com/NREL/resstock/blob/main/project_national/national_baseline.yml)
-  - v2.2.5 and prior: [Project folder openable in PAT](https://github.com/NREL/resstock/tree/v2.2.5/project_singlefamilydetached)
-- Unit-level OpenStudio Measures for automatically constructing OpenStudio Models of each representative dwelling unit model:
-  - v3.0.0 and later: [OpenStudio-HPXML Measures](https://github.com/NREL/resstock/tree/main/resources/hpxml-measures)
-  - v2.5.0 and prior: [OpenStudio Measures](https://github.com/NREL/resstock/tree/v2.5.0/resources/measures)
+- Project definition [buildstockbatch YML files](https://github.com/NREL/resstock/blob/main/project_national/national_baseline.yml) openable in any text editor for v2.3.0 and later
+- Unit-level [OpenStudio-HPXML Measures](https://github.com/NREL/resstock/tree/main/resources/hpxml-measures) (v3.0.0 and later) for automatically constructing individual representative dwelling unit OpenStudio Models using [OpenStudio-HPXML](https://github.com/NREL/OpenStudio-HPXML)
```
Comment (joseph-robertson) [inline README.md:16]: Note that these links are essentially duplicates of each other.

File: README.md [hunk 2]
```diff
@@ -12,15 +12,10 @@ This repository contains:
...
 - [Higher-level OpenStudio Measures](https://github.com/NREL/resstock/tree/main/measures) for controlling simulation inputs and outputs
 
 This repository does not contain software for running ResStock simulations, which can be found as follows:
```
Comment (joseph-robertson) [inline README.md:19]: I suppose this statement is technically not true anymore (we have run_analysis.rb).

Comment (afontani) [reply README.md]: I changed the language to include `run_analysis.rb`.

File: README.md (hunk 1 above)
Comment (afontani) [reply README.md]: I pointed to the docs of OpenStudio-HPXML for the second link.

**Review by joseph-robertson — APPROVED** [review]

**Pull request merged**